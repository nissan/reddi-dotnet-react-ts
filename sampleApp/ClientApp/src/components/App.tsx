import * as React from 'react';
import { Router } from 'react-router-dom';
import '../assets/styles/App.css';
import history from "../routing/history";
import Routes from "../routing/Routes";


class App extends React.Component {
  public render() {
    return (
      <Router history={history}>
        <Routes />
      </Router>
    );
  }
}

export default App;
