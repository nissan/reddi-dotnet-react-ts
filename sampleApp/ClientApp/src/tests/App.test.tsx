import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import App from "../components/App";


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});