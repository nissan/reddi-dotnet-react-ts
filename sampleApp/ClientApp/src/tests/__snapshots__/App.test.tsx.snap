// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src={undefined}
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.tsx
    </code>
     and save to reload.
  </p>
</div>
`;
